js-envs-test-kit/src/main/scala/org/scalajs/jsenv/test/TimeoutRunTests.scala

```diff
@@ -38,110 +38,21 @@ private[test] class TimeoutRunTests(config: JSEnvSuiteConfig, withCom: Boolean)
 
   }
 
-  @Test
-  def clearTimeoutTest: Unit = {
-
-    val deadline = 300.millis.fromNow
-
-    """
-    var c = setTimeout(function() { console.log("1"); }, 200);
-    setTimeout(function() {
-      console.log("2");
-      clearTimeout(c);
-    }, 100);
-    setTimeout(function() { console.log("3"); }, 300);
-    setTimeout(function() { console.log("4"); },   0);
-    """ hasOutput
-    """|4
-       |2
-       |3
-       |""".stripMargin
-
-    assertTrue("Execution took too little time", deadline.isOverdue())
-
-  }
-
-  @Test // #2368
-  def timeoutSingleArgTest: Unit = {
-    """
-    setTimeout(function() { console.log("ok"); });
-    """ hasOutput "ok\n"
-  }
-
-  @Test
-  def timeoutArgTest: Unit = {
-
-    val deadline = 300.millis.fromNow
-
-    """
-    setTimeout(function(a, b) { console.log("1" + a + b); }, 200, "foo", "bar");
-    setTimeout(function() { console.log("2"); }, 100);
-    setTimeout(function(msg) { console.log(msg); }, 300, "Hello World");
-    setTimeout(function() { console.log("4"); },   0);
-    """ hasOutput
-    """|4
-       |2
-       |1foobar
-       |Hello World
-       |""".stripMargin
-
-    assertTrue("Execution took too little time", deadline.isOverdue())
-
-  }
-
   @Test
   def intervalTest: Unit = {
-
-    val deadline = 1.second.fromNow
+    val deadline = 100.millis.fromNow
 
     """
-    var i1 = setInterval(function() { console.log("each 2200"); }, 2200);
-    var i2 = setInterval(function() { console.log("each 3100"); }, 3100);
-    var i3 = setInterval(function() { console.log("each 1300"); }, 1300);
-
-    setTimeout(function() {
-      clearInterval(i1);
-      clearInterval(i2);
-      clearInterval(i3);
-    }, 10000);
+    setInterval(function() { console.log("tick"); }, 20);
     """ hasOutput
```

Comment:
I'm totally confused by the new version of this test. How does it stop after 5 occurrences of the interval? The previous version explicitly called `clearInterval` after some time, which explained the behavior. But now, what makes it stop?

Reply:
The kit makes it stop :) We cannot rely anymore on `JSEnvs` terminating. So once we have seen the output we need, we terminate it.

Reply:
Oh! Maybe add a comment to remind the reader about that in this specific test? Because in this particular case it's non-obvious why the test is even correct if the reader doesn't have that property in mind. (In other tests I've read, it's not a problem for understanding the test.)